lib/config-generator.js

```diff
@@ -155,67 +117,130 @@ export class ConfigGenerator {
-            exportContent += "  tseslint.configs.recommended,\n";
+        // language = javascript/typescript
+        if (languages.includes("javascript")) {
+            if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
+                exportContent += `  { files: ["**/*.js"], languageOptions: { sourceType: "${this.answers.moduleType}" } },\n`;
+            }
+
+            if (this.answers.env?.length > 0) {
+                this.result.devDependencies.push("globals");
+                importContent += "import globals from \"globals\";\n";
+                const envContent = {
+                    browser: "globals: globals.browser",
+                    node: "globals: globals.node",
+                    "browser,node": "globals: {...globals.browser, ...globals.node}"
+                };
+
+                exportContent += `  { files: ["${extensions}"], languageOptions: { ${envContent[this.answers.env.join(",")]} } },\n`;
+            }
+            if (useTs) {
+                this.result.devDependencies.push("typescript-eslint");
+                importContent += "import tseslint from \"typescript-eslint\";\n";
+                exportContent += "  tseslint.configs.recommended,\n";
+            }
+
+            if (this.answers.framework === "vue") {
+                this.result.devDependencies.push("eslint-plugin-vue");
+                importContent += "import pluginVue from \"eslint-plugin-vue\";\n";
+                exportContent += "  pluginVue.configs[\"flat/essential\"],\n";
+
+                // https://eslint.vuejs.org/user-guide/#how-to-use-a-custom-parser
+                if (useTs) {
+                    exportContent += "  { files: [\"**/*.vue\"], languageOptions: { parserOptions: { parser: tseslint.parser } } },\n";
+                }
+            }
+
+            if (this.answers.framework === "react") {
+                this.result.devDependencies.push("eslint-plugin-react");
+                importContent += "import pluginReact from \"eslint-plugin-react\";\n";
+                exportContent += "  pluginReact.configs.flat.recommended,\n";
+            }
+
+            if (needCompatHelper) {
+                this.result.devDependencies.push("@eslint/eslintrc", "@eslint/js");
+            }
+        } else {
+            exportContent += "  { ignores: [\"**/*.js\", \"**/*.cjs\", \"**/*.mjs\"] },\n";
+        }
+
+        // language = json/jsonc/json5
+        if (languages.some(item => item.startsWith("json"))) {
+            this.result.devDependencies.push("@eslint/json");
+            importContent += "import json from \"@eslint/json\";\n";
+
+            if (languages.includes("json")) {
+                const config = purpose === "syntax"
+                    ? "  { files: [\"**/*.json\"], plugins: { json }, language: \"json/json\" },\n"
+                    : "  { files: [\"**/*.json\"], language: \"json/json\", extends: [\"json/recommended\"],\n";
```

Comment:
```suggestion
                    : "  { files: [\"**/*.json\"], plugins: { json }, language: \"json/json\", extends: [\"json/recommended\"] },\n";
```

Missing closing `}`, and we need to add the plugin here.